Reminder-App/Reminder-App/ReminderDetailViewController.swift

```diff
@@ -9,62 +9,39 @@ import UIKit
 
 class ReminderDetailViewController: UITableViewController {
     
-    enum ReminderRow: Int, CaseIterable {
-        case title
-        case date
-        case time
-        case notes
-        
-        func displayText(for reminder: Reminder?) -> String? {
-            switch self {
-            case .title:
-                return reminder?.title
-            case .date:
-                return reminder?.dueDate.description
-            case .time:
-                return reminder?.dueDate.description
-            case .notes:
-                return reminder?.notes
-            }
-        }
-        
-        var cellImage: UIImage? {
-            switch self {
-            case .title:
-                return nil
-            case .date:
-                return UIImage(systemName: Images.calenderCirceImage)
-            case .time:
-                return UIImage(systemName: Images.clockImage)
-            case .notes:
-                return UIImage(systemName: Images.squareAndPencil)
-            }
-        }
-    }
-    
-    var reminder: Reminder?
+    // MARK: Private Propeeties
+    private var reminder: Reminder?
+    private var dataSource: UITableViewDataSource?
     
+    // MARK: Instance Methods
     func configure(with reminder: Reminder) {
         self.reminder = reminder
     }
     
-}
-
-extension ReminderDetailViewController {
-    
-    static let reminderDetailCellIdentifier = "ReminderDetailCell"
-    
-    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
-        return ReminderRow.allCases.count
+    // MARK:- Lifecycle Methods
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        setEditing(false, animated: false)
+        navigationItem.setRightBarButton(editButtonItem, animated: false)
+        tableView.register(UITableViewCell.self, forCellReuseIdentifier: ReminderDetailEditDataSource.dateLabelCellIdentifier)
     }
     
-    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
-        let cell = tableView.dequeueReusableCell(withIdentifier: Self.reminderDetailCellIdentifier, for: indexPath)
-        let row = ReminderRow(rawValue: indexPath.row)
+    // MARK: Overridden Methods
+    override func setEditing(_ editing: Bool, animated: Bool) {
+        super.setEditing(editing, animated: animated)
         
-        cell.textLabel?.text = row?.displayText(for: reminder)
-        cell.imageView?.image = row?.cellImage
+        guard let reminder = reminder else {
+            fatalError("No reminder found for detail view")
```

Comment:
same as above. the purpose of unwrapping is to avoid crash when there are unexpected nil values